### PR TITLE
fix: Raise on oob decimal precision

### LINKED ISSUE #17425: Conversion to pyarrow crashes the interpreter with a core dump when using decimals
### Checks

- [X] I have checked that this issue has not already been reported.
- [X] I have confirmed this bug exists on the [latest version](https://pypi.org/project/polars/) of Polars.

### Reproducible example

```
In [9]: import polars as pl

In [10]: pl.__version__
Out[10]: '1.0.0'

In [11]: df = pl.DataFrame({"a": [1.0]})

In [12]: df
Out[12]:
shape: (1, 1)
┌─────┐
│ a   │
│ --- │
│ f64 │
╞═════╡
│ 1.0 │
└─────┘

In [13]: df = df.select(b=pl.col("a").cast(pl.Decimal(76, 38)))

In [14]: df
Out[14]:
shape: (1, 1)
┌─────────────────────────────────┐
│ b                               │
│ ---                             │
│ decimal[76,38]                  │
╞═════════════════════════════════╡
│ 0.9999999999999999774880982345… │
└─────────────────────────────────┘

In [15]: df.to_arrow()
/arrow/cpp/src/arrow/type.cc:1466:  Check failed: (precision) <= (kMaxPrecision)
zsh: abort (core dumped)  ipython
```

### Log output

_No response_

### Issue description

Polars dumps core when casting decimals.

### Expected behavior

I would expect an error message, or a successful conversion to pyarrow, really anything but dumping core and crashing the Python interpreter.

### Installed versions

<details>

```
--------Version info---------
Polars:               1.0.0
Index type:           UInt32
Platform:             Linux-6.6.36-x86_64-with-glibc2.39
Python:               3.10.14 (main, Mar 19 2024, 21:46:16) [GCC 13.3.0]

----Optional dependencies----
adbc_driver_manager:  <not installed>
cloudpickle:          3.0.0
connectorx:           <not installed>
deltalake:            0.18.1
fastexcel:            <not installed>
fsspec:               2024.6.1
gevent:               <not installed>
great_tables:         <not installed>
hvplot:               <not installed>
matplotlib:           3.9.0
nest_asyncio:         1.6.0
numpy:                2.0.0
openpyxl:             <not installed>
pandas:               2.2.2
pyarrow:              16.1.0
pydantic:             2.7.4
pyiceberg:            <not installed>
sqlalchemy:           2.0.31
torch:                <not installed>
xlsx2csv:             <not installed>
xlsxwriter:           <not installed>
```

</details>

### PR DESCRIPTION
**Pull request by ritchie46**
fixes #17425

### CODE REVIEW

File: crates/polars-core/src/datatypes/dtype.rs [hunk 1]
```diff
@@ -542,11 +543,15 @@ impl DataType {
             Float32 => Ok(ArrowDataType::Float32),
             Float64 => Ok(ArrowDataType::Float64),
             #[cfg(feature = "dtype-decimal")]
-            // note: what else can we do here other than setting precision to 38?..
-            Decimal(precision, scale) => Ok(ArrowDataType::Decimal(
-                (*precision).unwrap_or(38),
-                scale.unwrap_or(0), // and what else can we do here?
-            )),
+            Decimal(precision, scale) => {
```
Comment (stinodego) [inline crates/polars-core/src/datatypes/dtype.rs:546]: We should probably also check that `scale <= precision`.

Comment (ritchie46) [reply crates/polars-core/src/datatypes/dtype.rs:546]: I don't believe scale is limited: https://arrow.apache.org/docs/python/generated/pyarrow.decimal128.html

Comment (cpcloud) [inline crates/polars-core/src/datatypes/dtype.rs:546]: PyArrow supports decimal256 too, is there any reason this should error as opposed to outputting a decimal256 array?

Comment (ritchie46) [reply crates/polars-core/src/datatypes/dtype.rs:546]: We don't support `Decimal256`. It turned out the precision given wasn't supported by Polars, so we raise earlier. Therefore we would also never need to `Decimal256`.

**Pull request merged**